lib/Versions/ExpireManager.php

```diff
@@ -66,8 +66,11 @@ protected function getAutoExpireList(int $time, array $versions): array {
 			while ($newInterval) {
 				if ($nextInterval === -1 || $prevTimestamp > $nextInterval) {
 					if ($version->getTimestamp() > $nextVersion) {
-						//distance between two version too small, mark to delete
-						$toDelete[] = $version;
+						// Do not expire versions with a label.
+						if ($version->getMetadataValue('label') === null || $version->getMetadataValue('label') === '') {
```

Comment:
Should never occur, but makes it obvious and makes psalm happy

```suggestion
						if (!($version instanceof IMetadataVersion) || $version->getMetadataValue('label') === null || $version->getMetadataValue('label') === '') {
```

Reply:
Not sure how it fits with the 'auto expire' logic. This probably retain more version than wanted, but prevents data loss. So fine with me.

Reply:
Depending on the specific case, more versions will be retained. But that's the intention if a version is labeled, right? But expectations could be different.
Actually, while doing initial tests with this additional check for `IMetadataVersion`, versions were not retained as expected. Thus, I will do some more tests.